Tasks/Common/java-common/java-common.ts

```diff
@@ -55,3 +56,17 @@ export function findJavaHome(jdkVersion: string, jdkArch: string): string {
     
     return discoveredJavaHome;
 }
+
+export function publishJavaTelemetry(taskName: string, javaTelemetryData) {
+    try {
+        //tl.assertAgent('2.120.0'); -> we can use this when all the tasks using this common module use vsts-task-lib 2.1.0 or higher
+        let agentVersion: string = tl.getVariable('Agent.Version');
+        if (agentVersion && !semver.lt(agentVersion, '2.120.0') && taskName && javaTelemetryData) {
+            console.log('##vso[telemetry.publish area=Tasks.CrossPlatform;feature=' + taskName + ']' + JSON.stringify(javaTelemetryData));
+        } else {
+            tl.debug('Failed to publish java telemetry. Agent version 2.120.0 or higher is required.');
```

Comment:
This is probably fine.  Just want to mention that this is not a "failure" per se, this is more like "Skipping publish java telemetry".  People are sensitive to works like "Failed" | "Error" | "Warning" in logs.  :)

Reply:
Good point, will update when we switch to tl.assertAgent()